code/controllers/subsystem/processing/quirks.dm

```diff
@@ -137,19 +138,22 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		quirks_to_add += quirk
 
 	//And have benefits too
-	while(score < 0 && good_count <= MAX_QUIRKS)
-		if(!length(possible_quirks))//Lets not get stuck
-			break
-		var/quirk = pick(quirks)
-		if(quirk in GLOB.quirk_blacklist) //prevent blacklisted
-			possible_quirks -= quirk
-			continue
-		if(!quirk_points[quirk] > 0) //positive only
-			possible_quirks -= quirk
-			continue
-		good_count++
-		score += quirk_points[quirk]
-		quirks_to_add += quirk
+	if(max_positive_quirks < 0)
+		max_positive_quirks = 6
```

Comment:
Nit: Do this closer to the definition of max_positive_quirks

Reply:
Fixed by commit [74b561c](https://github.com/tgstation/tgstation/pull/81033/commits/74b561cdc7435eae66da2f589f41ef6c5c3fdcc5)